### PR TITLE
feat: Add enterprise custom properties

### PR DESCRIPTION
**Pull request by stevehipwell**
This PR adds enterprise scoped [custom properties](https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/custom-properties?apiVersion=2022-11-28).

> [!NOTE]
> This API endpoint is currently still in preview.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @stevehipwell - this looks great!
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin !
Merging.

**Pull request merged**
